py-polars/tests/unit/io/test_database_write.py

```diff
@@ -11,93 +12,124 @@
-            ),
-        ),
-        pytest.param(
-            "sqlalchemy",
-            "create",
-            id="sa_create",
-        ),
-        pytest.param(
-            "sqlalchemy",
-            "append",
-            id="sa_append",
-        ),
-    ],
+@pytest.mark.parametrize("engine", ["adbc", "sqlalchemy"])
+@pytest.mark.skipif(
+    sys.version_info < (3, 9) or sys.platform == "win32",
+    reason="adbc_driver_sqlite not available below Python 3.9 / on Windows",
 )
-def test_write_database(
-    engine: DbWriteEngine, mode: DbWriteMode, sample_df: pl.DataFrame, tmp_path: Path
-) -> None:
-    tmp_path.mkdir(exist_ok=True)
-    tmp_db = f"test_{engine}.db"
-    test_db = str(tmp_path / tmp_db)
+def test_write_database_append(engine: DbWriteEngine, tmp_path: Path) -> None:
+    df = pl.DataFrame(
+        {
+            "key": ["xx", "yy", "zz"],
+            "value": [123, None, 789],
+            "other": [5.5, 7.0, None],
+        }
+    )
 
-    # note: test a table name that requires quotes to ensure that we handle
-    # it correctly (also supply an explicit db schema with/without quotes)
-    table_name = "test_data"
+    tmp_path.mkdir(exist_ok=True)
+    test_db = str(tmp_path / f"test_{engine}.db")
+    table_name = "test_append"
 
-    sample_df.write_database(
+    df.write_database(
         table_name=table_name,
         connection=f"sqlite:///{test_db}",
         if_exists="replace",
         engine=engine,
     )
-    if mode == "append":
-        sample_df.write_database(
+
+    ExpectedError = NotImplementedError if engine == "adbc" else ValueError
+    with pytest.raises(ExpectedError):
+        df.write_database(
             table_name=table_name,
             connection=f"sqlite:///{test_db}",
-            if_exists="append",
+            if_exists="fail",
             engine=engine,
         )
-        sample_df = pl.concat([sample_df, sample_df])
 
+    df.write_database(
+        table_name=table_name,
+        connection=f"sqlite:///{test_db}",
+        if_exists="append",
+        engine=engine,
+    )
     result = pl.read_database_uri(f"SELECT * FROM {table_name}", f"sqlite:///{test_db}")
-    sample_df = sample_df.with_columns(pl.col("date").cast(pl.Utf8))
-    assert_frame_equal(sample_df, result)
+    assert_frame_equal(result, pl.concat([df, df]))
+
+
+@pytest.mark.write_disk()
+@pytest.mark.parametrize("engine", ["adbc", "sqlalchemy"])
+@pytest.mark.skipif(
+    sys.version_info < (3, 9) or sys.platform == "win32",
+    reason="adbc_driver_sqlite not available below Python 3.9 / on Windows",
+)
+def test_write_database_create_quoted_tablename(
+    engine: DbWriteEngine, tmp_path: Path
+) -> None:
+    if engine == "adbc" and adbc_driver_sqlite.__version__ == "0.6.0":  # type: ignore[attr-defined]
+        # see: https://github.com/apache/arrow-adbc/issues/1000
+        pytest.skip(
+            reason="ADBC SQLite driver v0.6.0 has a bug with quoted/qualified table names"
+        )
```

Comment:
This can also just be a `@pytest.mark.skipif`, right?

Reply:
Ahh, can you can reference the value of a parameterised param in the skipIf? If so, then "yes", but I didn't realise you could do that (we still want the test to run for sqlalchemy).

Reply:
Oh, maybe you can't... wanna find out? :smile:

Reply:
> Oh, maybe you can't... wanna find out? 😄

Lol... well, when I (very briefly!) checked, the StackOverflow suggestion was an inline `pytest.skip` 😅